modules/search/src/main/java/org/opencastproject/matterhorn/search/impl/AbstractElasticsearchQueryBuilder.java

```diff
@@ -45,33 +40,26 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Date;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
  * Opencast implementation of the elastic search query builder.
  */
-public abstract class AbstractElasticsearchQueryBuilder<T extends SearchQuery> implements QueryBuilder {
+public abstract class AbstractElasticsearchQueryBuilder<T extends SearchQuery> extends QueryBuilder {
 
   /** Term queries on fields */
   private Map<String, Set<Object>> searchTerms = null;
 
-  /** Negative term queries on fields */
```

Comment:
@lkiesow Could you please provide a few words about why negative term queries, empty fields and non-empty fields are not required (anymore)?

Reply:
Hm, it has been a while and a Christmas break does not help remembering :-D

But if I remember correctly then this code would need to be written differently since the API had changed and when I looked into this, I noticed that Opencast was never actually using this at all. Hence it seemed like the sensible way to remove this instead of having dead code lying around.